etl/tests/failpoints/pipeline_test.rs

```diff
@@ -0,0 +1,242 @@
+use etl::destination::memory::MemoryDestination;
+use etl::failpoints::START_TABLE_SYNC_AFTER_DATA_SYNC;
+use etl::pipeline::{PipelineError, PipelineId};
+use etl::state::store::notify::NotifyingStateStore;
+use etl::state::table::TableReplicationPhaseType;
+use etl::workers::base::WorkerWaitError;
+use fail::FailScenario;
+use rand::random;
+use telemetry::init_test_tracing;
+
+use etl::test_utils::database::spawn_database;
+use etl::test_utils::pipeline::create_pipeline;
+use etl::test_utils::test_destination_wrapper::TestDestinationWrapper;
+use etl::test_utils::test_schema::{TableSelection, setup_test_database_schema};
+
+/*
+Tests that we want to add:
+TBD
+ */
+
+#[tokio::test(flavor = "multi_thread")]
+async fn pipeline_handles_table_sync_worker_panic_during_data_sync() {
```

Comment:
This is a simple test which uses `fail-rs` to prove that it's working. More complex tests will be added.